kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,321 @@
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.networkanalysis.Route
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteTask
+import com.esri.arcgisruntime.tasks.networkanalysis.Stop
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.bottom_sheet.view.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  private val bottomSheetBehavior: BottomSheetBehavior<View> by lazy {
+    // create a bottom sheet behavior from the view included in the main layout
+    BottomSheetBehavior.from(bottomSheet)
+  }
+
+  private val graphicsOverlay: GraphicsOverlay by lazy {
+    // create a graphics overlay and add it to the map view
+    GraphicsOverlay().also {
+      mapView.graphicsOverlays.add(it)
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create new basemap with the vector tiled layer from a service url
+    val basemap = Basemap(
+      ArcGISVectorTiledLayer(resources.getString(R.string.navigation_vector))
+    )
+    // create a map with the basemap
+    ArcGISMap(basemap).let { map ->
+      // set initial viewpoint to San Diego
+      map.initialViewpoint = Viewpoint(32.7157, -117.1611, 200000.0)
+      // set the map to be displayed in this view
+      mapView.map = map
+    }
+
+    // ensure the floating action button moves to be above the attribution view
+    val params = directionFab.layoutParams as ConstraintLayout.LayoutParams
+    mapView.addAttributionViewLayoutChangeListener { _, _, _, _, bottom, _, _, _, oldBottom ->
+      val heightDelta = bottom - oldBottom
+      params.bottomMargin += heightDelta
+    }
+
+    // hide the bottom sheet
+    bottomSheetBehavior.peekHeight = 0
+    // animate the arrow when the bottom sheet slides
+    bottomSheetBehavior.addBottomSheetCallback(object : BottomSheetCallback() {
+      override fun onSlide(bottomSheet: View, slideOffset: Float) {
+        bottomSheet.header.imageView.rotation = slideOffset * 180f
+      }
+
+      override fun onStateChanged(bottomSheet: View, newState: Int) {
+        bottomSheet.header.imageView.rotation = when (newState) {
+          BottomSheetBehavior.STATE_EXPANDED -> 180f
+          else -> 0f
+        }
+      }
+    })
+    // while the bottom sheet is hidden, make the map view span the whole screen
+    val mainContainerParams = mainContainer.layoutParams as CoordinatorLayout.LayoutParams
+    mainContainerParams.bottomMargin = 0
```

Comment:
I think a lot of this really neat UI should be hidden away in a method where possible: we want the main code focus to be on our API code which should be first and foremost in the sample. If we can tuck this UI logic away into a method and just call that method here, that will be a lot more tidy and allow customers to hone in on our API code more easily :)

Reply:
You are 100% right! Will address in my pr comment